articles/grand-rename.md

```diff
@@ -0,0 +1,97 @@
+
+### Signals
+
+- `Session`: Anything related to starting, pausing, continuing, or ending a user session.
+
+### Parameters
+
+- `AppContext`: The context the app runs in, such as the locale, the debug mode, or the target environment.
+- `AppInfo`: Information about the specific app build, such as version, build number, or SDKs compiled with.
+- `Device`: All about the device running the application, such as operating system, model name, or architecture.
+- `SDK`: Information about the TelemetryDeck SDK, such as the version number or supported features.
+
+{% noteinfo "No action needed" %}
+To make sure these changes don't affect existing customers with existing insights anytime soon, we decided to keep sending the old names for a transition period of at least 1 year while also sending the new ones. We have also auto-migrated any insights that were using the old names in their filters to work with the new system. This was safely achieved by replacing filters like "platform == iOS" with something like "platform == iOS OR TelemetryDeck.SystemInfo.platform == iOS" accepting both styles, ensuring you will continue to see historic data from before the transition, during the transition, and after the transition.
+{% endnoteinfo %}
+
+<!-- Discussion: What about Top-N insights that use a parameter like "appVersion" which got renamed to "TelemetryDeck.SystemInfo.appVersion"? -->
+
+## Filling the Gaps
+
+While doing the rename, we also noticed some missing variants of parameters that could be useful. So we added the following variants:
+
+- `TelemetryDeck.AppContext.region`: The regional identifier of the users country setting, e.g. `US` for United States. Used for things like date format.
+- `TelemetryDeck.AppContext.language`: The language the app is currently used in (one of the supported languages).
+- `TelemetryDeck.AppInfo.versionAndBuildNumber`: Combines the app version and build into one String, such as `1.7.1 (build 22)`.
+- `TelemetryDeck.Device.preferredLanguage`: The language most preferred by the user on this device (might not be supported by app).
+- `TelemetryDeck.Device.type`: One of `Phone`, `Tablet`, `Desktop`, `Watch`, `TV`, or `Headset`
+- `TelemetryDeck.Device.screenResolutionWidth`: The resolution width of the screen in pixel/points (whatever is most common on platform).
+- `TelemetryDeck.Device.screenResolutionHeight`: The resolution height of the screen in pixel/points (whatever is most common on platform).
+- `TelemetryDeck.Device.orientation`: One of `Portrait`, `Landscape`, or `Fixed`.
+- `TelemetryDeck.Device.timeZone`: The timezone expressed by the UTC offset, such as `UTC+0` (London), `UTC+9` (Tokyo), or `UTC-8` (San Francisco).
+- `TelemetryDeck.SDK.supportsPrefixes`: Indicates if the TelemetryDeck SDK version is already migrated to the new prefixed naming system.
```

Comment:
Can't we derive that form `TelemetryDeck.SDK.version`?

Reply:
Oh and speaking of stuff we can put under the SDK namespace: 

`TelemetryDeck.SDK.version` could be just the version string, with a separate parameter for the SDK ... platform, like `TelemetryDeck.SDK.platform`. Not sure if platform is the correct naming here, do you have better suggestions?

```json
{
  "TelemetryDeck.SDK.version": "2.0.0",
  "TelemetryDeck.SDK.platform": "SwiftSDK"
}
```

Reply:
I think TelemetryDeck.SDK.platform is quite good, only alternative that comes to my mind is TelemetryDeck.SDK.framework if you want to about platform.

Reply:
On second thought `platform` doesn't seem quite right given that most SDKs are named by programming languages. If we go by `Platform`, the Kotlin SDK should be renamed to `AndroidSDK` and the Swift one to `AppleSDK`, for example. But I think `platform` is too easy to misunderstand and confuse with `TelemetryDeck.Device.platform`, so we should avoid it.

For `SwiftSDK` / `KotlinSDK` / `FlutterSDK`, we could use `language`, `variant`, `type`, `id`, or simply `name`. I chatted a bit with ChatGPT and we figured `name` is probably the most flexible and clear one. Let me know if you disagree, I don't have a strong feeling about the naming here.

Reply:
I just split the old `telemetryClientVersion` up to `TelemetryDeck.SDK.name` and `TelemetryDeck.SDK.version`, but for compatibility reasons I kept the old style and renamed that to `TelemetryDeck.SDK.nameAndVersion`. See 1f2f3a5e31f4ced1120e2ab13ed6dfa483b7fe26.